packages/react-dom/src/shared/ReactDOMUnknownPropertyHook.js

```diff
@@ -226,6 +227,25 @@ if (__DEV__) {
       return false;
     }
 
+    // Warn when passing the string 'false' into a boolean prop
+    if (
+      propertyInfo !== null &&
+      propertyInfo.type === BOOLEAN &&
+      typeof value === 'string' &&
+      value.toLowerCase() === 'false'
+    ) {
```

Comment:
Let’s make `value === 'false'` the first condition. Since it’s rare we won’t have to check most other conditions.